libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -704,4 +736,74 @@ std::shared_ptr<arrow::DataType> make_experimental_type(const type& t) {
   return caf::visit(f, t);
 }
 
+type make_vast_type(const std::shared_ptr<arrow::DataType>& arrow_type) {
+  switch (arrow_type->id()) {
+    case arrow::Type::NA:
+      return type{none_type{}};
+    case arrow::Type::BOOL:
+      return type{bool_type{}};
+    case arrow::Type::INT64:
+      return type{integer_type{}};
+    case arrow::Type::UINT64:
+      return type{count_type{}};
+    case arrow::Type::DOUBLE:
+      return type{real_type{}};
+    case arrow::Type::DURATION:
+      return type{duration_type{}};
+    case arrow::Type::STRING:
+      return type{string_type{}};
+    case arrow::Type::TIMESTAMP:
```

Comment:
If I understood correctly then the Arrow Timestamp data type can have different resolutions from nanoseconds. You probably need to check whether this is a nanosecond-resolution timestamp, and fail otherwise.

The same applies to duration.

Reply:
good catch, additional inspection of the time unit is necessary, even though for VAST-generated data this should already be the right time unit.